### DISCUSSION TITLE
Number input with 0 precision ?

### DISCUSSION THREAD

**Discussion by jekoie** [zauberzeug/nicegui#2580] [Q&A]
### Description

The `ui.number` with 0 precision should be an integer value when input some value, but it give me a float value with 1 precision.

**demo code**
```python
from nicegui import app, ui
ui.number("Count:", precision=0).bind_value(app.storage.general, "count")
ui.label().bind_text_from(app.storage.general, "count", lambda v: f"{v}")
ui.run()

**Comment by falkoschindler** [comment]
Hi @jekoie!

As far as I can tell the `ui.number` elements shows the number "0" without any decimal places, just as expected.
The `ui.label`, on the other hand, shows "0.0". This is also expected, because the label isn't influenced by the `precision` argument. Instead it's the `f"{v}"` format string which defines the format for the label text. And a float value of 0.0 is displayed as "0.0" by default. You can explicitly remove the decimal places with `f"{v:.0f}"`.

**Comment by jekoie** [comment]
When I change `ui.number` value, it always be  `int.0`, the 0 precision tell me whatever value change, it shoud be `int` value. The format `f"{v:.0f}"` just change display, indeed the real value is a float value with 1 precision with a small point zero.

**Comment by jekoie** [comment]
So I want a integer value when `ui.number` change, How can I do ?

**Reply by HansBambel** [reply]
I also needed to be the value it writes to to be an integer. A way to achieve that is to use `forward=int`:
```python
ui.number("Count:", precision=0).bind_value(app.storage.general, "count", forward=int)
```

**Comment by falkoschindler** [comment]
@jekoie The `precision` argument controls the number of digits that are _displayed_ within the `ui.number` element. The value is still a float. Depending on what you want to do with the value, you can simply convert it with `int(...)` or render it accordingly with `"{v:.0f}"`.
